pceplib/pcep_msg_objects_encoding.c

```diff
@@ -0,0 +1,1483 @@
+struct pcep_object_header *pcep_decode_obj_inter_layer(struct pcep_object_header *hdr, const uint8_t *buf);
+struct pcep_object_header *pcep_decode_obj_switch_layer(struct pcep_object_header *hdr, const uint8_t *buf);
+struct pcep_object_header *pcep_decode_obj_req_adap_cap(struct pcep_object_header *hdr, const uint8_t *buf);
+struct pcep_object_header *pcep_decode_obj_server_ind(struct pcep_object_header *hdr, const uint8_t *buf);
+struct pcep_object_header *pcep_decode_obj_objective_function(struct pcep_object_header *hdr, const uint8_t *buf);
+typedef struct pcep_object_header* (*object_decoder_funcptr)(struct pcep_object_header *, const uint8_t *buf);
+
+object_decoder_funcptr object_decoders[MAX_OBJECT_ENCODER_INDEX];
+
+/* Object lengths, including the Object Header.
+ * Used by pcep_object_get_length() and pcep_object_has_tlvs() */
+static uint8_t pcep_object_class_lengths[] = {
+        0,   /* Object class 0 unused */
+        8,   /* PCEP_OBJ_CLASS_OPEN = 1 */
+        12,  /* PCEP_OBJ_CLASS_RP = 2 */
+        16,  /* PCEP_OBJ_CLASS_NOPATH = 3, includes 8 for mandatory TLV */
+        0,   /* PCEP_OBJ_CLASS_ENDPOINTS = 4, could be ipv4 or ipv6, setting to 0 */
+        8,   /* PCEP_OBJ_CLASS_BANDWIDTH = 5 */
+        12,  /* PCEP_OBJ_CLASS_METRIC = 6 */
+        0,   /* PCEP_OBJ_CLASS_ERO = 7, setting 0, ROs cannot have TLVs */
+        0,   /* PCEP_OBJ_CLASS_RRO = 8, setting 0, ROs cannot have TLVs */
+        20,  /* PCEP_OBJ_CLASS_LSPA = 9 */
+        0,   /* PCEP_OBJ_CLASS_IRO = 10, setting 0, ROs cannot have TLVs */
+        0,   /* PCEP_OBJ_CLASS_SVEC = 11, SVECs cannot have TLVs */
+        8,   /* PCEP_OBJ_CLASS_NOTF = 12 */
+        8,   /* PCEP_OBJ_CLASS_ERROR = 13 */
+        0,   /* Object class 14 unused */
+        8,   /* PCEP_OBJ_CLASS_CLOSE = 15 */
+        0, 0, 0, 0, 0, /* Object classes 16 - 20 are not used */
+        8,   /* PCEP_OBJ_CLASS_OF = 21 */
+        0, 0, 0, 0, 0,   /* Object classes 22 - 26 are not used */
+        0, 0, 0, 0, 0,   /* Object classes 27 - 31 are not used */
+        8,   /* PCEP_OBJ_CLASS_LSP = 32 */
+        12,  /* PCEP_OBJ_CLASS_SRP = 33 */
+        12,  /* PCEP_OBJ_CLASS_VENDOR_INFO = 34 */
+        0,   /* Object class 35 unused */
+        0,   /* PCEP_OBJ_CLASS_INTER_LAYER = 36, cannot have TLVs */
+        0,   /* PCEP_OBJ_CLASS_SWITCH_LAYER = 37, cannot have TLVs */
+        0,   /* PCEP_OBJ_CLASS_REQ_ADAP_CAP = 38, cannot have TLVs*/
+        8,   /* PCEP_OBJ_CLASS_SERVER_IND = 39 */
+        0,   /* PCEP_OBJ_CLASS_ASSOCIATION = 40, cannot have TLVs */
+};
+
+
+static void initialize_object_coders()
+{
+    static bool initialized = false;
+
+    if (initialized == true)
+    {
+        return;
+    }
+
+    initialized = true;
+
+    /* Encoders */
+    memset(object_encoders, 0, sizeof(object_encoder_funcptr) * MAX_OBJECT_ENCODER_INDEX);
+    object_encoders[PCEP_OBJ_CLASS_OPEN]       =  pcep_encode_obj_open;
```

Comment:
Unless I'm missing something, all of this can be assigned statically and there is no need for a function here. See `zserv_handlers` in `zebra/zapi_msg.c` for an example.

Reply:
done